app/src/test/java/com/github/alfabravo2013/readyforexams/domain/login/LoginUseCaseTest.kt

```diff
@@ -0,0 +1,78 @@
+package com.github.alfabravo2013.readyforexams.domain.login
+
+import com.github.alfabravo2013.readyforexams.util.Result
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+
+internal class LoginUseCaseTest {
+    private val repository = mockk<LoginRepository>()
+    private val loginUseCase = LoginUseCase(repository)
+
+    private val registeredEmail = "test@test.com"
+    private val unregisteredEmail = "unknown@test.com"
+    private val correctPassword = "123456789"
+
+    @BeforeEach
+    fun setup() {
+        every { repository.login(registeredEmail, correctPassword) } returns Result.Success
```

Comment:
based on your situations you should not mock everything in BeforeEach. Ideally mocking for your test method should be done inside the test. There are situations where you can mock something that applies to all your tests. In this situations I would suggest to mock success result inside the success test and failure when you test that.